cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1160,88 +1162,85 @@ void writer::impl::write_chunk(table_view const &table, pq_chunked_state &state)
-                                   state.stream.value()));
-          state.stream.synchronize();
+                                   stream.value()));
+          stream.synchronize();
           out_sink_->host_write(host_bfr.get() + ck->ck_stat_size, ck->compressed_size);
           if (ck->ck_stat_size != 0) {
-            state.md.row_groups[global_r].columns[i].meta_data.statistics_blob.resize(
-              ck->ck_stat_size);
-            memcpy(state.md.row_groups[global_r].columns[i].meta_data.statistics_blob.data(),
+            md.row_groups[global_r].columns[i].meta_data.statistics_blob.resize(ck->ck_stat_size);
+            memcpy(md.row_groups[global_r].columns[i].meta_data.statistics_blob.data(),
                    host_bfr.get(),
                    ck->ck_stat_size);
           }
         }
-        state.md.row_groups[global_r].total_byte_size += ck->compressed_size;
-        state.md.row_groups[global_r].columns[i].meta_data.data_page_offset =
-          state.current_chunk_offset + ((ck->has_dictionary) ? ck->dictionary_size : 0);
-        state.md.row_groups[global_r].columns[i].meta_data.dictionary_page_offset =
-          (ck->has_dictionary) ? state.current_chunk_offset : 0;
-        state.md.row_groups[global_r].columns[i].meta_data.total_uncompressed_size = ck->bfr_size;
-        state.md.row_groups[global_r].columns[i].meta_data.total_compressed_size =
-          ck->compressed_size;
-        state.current_chunk_offset += ck->compressed_size;
+        md.row_groups[global_r].total_byte_size += ck->compressed_size;
+        md.row_groups[global_r].columns[i].meta_data.data_page_offset =
+          current_chunk_offset + ((ck->has_dictionary) ? ck->dictionary_size : 0);
+        md.row_groups[global_r].columns[i].meta_data.dictionary_page_offset =
+          (ck->has_dictionary) ? current_chunk_offset : 0;
+        md.row_groups[global_r].columns[i].meta_data.total_uncompressed_size = ck->bfr_size;
+        md.row_groups[global_r].columns[i].meta_data.total_compressed_size   = ck->compressed_size;
+        current_chunk_offset += ck->compressed_size;
       }
     }
   }
 }
 
-std::unique_ptr<std::vector<uint8_t>> writer::impl::write_chunked_end(
-  pq_chunked_state &state, bool return_filemetadata, const std::string &column_chunks_file_path)
+std::unique_ptr<std::vector<uint8_t>> writer::impl::close(
+  std::string const &column_chunks_file_path)
 {
+  closed = true;
   CompactProtocolWriter cpw(&buffer_);
   file_ender_s fendr;
   buffer_.resize(0);
-  fendr.footer_len = static_cast<uint32_t>(cpw.write(state.md));
+  fendr.footer_len = static_cast<uint32_t>(cpw.write(md));
   fendr.magic      = parquet_magic;
   out_sink_->host_write(buffer_.data(), buffer_.size());
   out_sink_->host_write(&fendr, sizeof(fendr));
   out_sink_->flush();
 
   // Optionally output raw file metadata with the specified column chunk file path
-  if (return_filemetadata) {
+  if (column_chunks_file_path.length() > 0) {
```

Comment:
It appears that in the old code, it was valid to pass `return_filemetadata=true` but keep the `column_chunks_file_path` empty. The only difference that would make is that the returned metadata would not have the `metadata.row_groups.columns[i].file_path` set. With the new change, the `column_chunks_file_path` would need to have a value in order to return the metadata blob.

Coupled with the cython interface, this relationship between `return_filemetadata` and `column_chunks_file_path` makes sense but libcudf should sense on its own. You could either change the parameter names or the docs.

Reply:
I will update the doc to to expect column chunks file path is expected to return filemetadata.